### PR TITLE
Add support for Swift 5.5 and async/await

### PR DESCRIPTION
**Pull request by ptoffy**
Updated `swift-tools-version` to 5.5 and added `async/await` support to the project. 
Removed `EventLoopFuture` tests as they are redundant when using `async/await`.

### CODE REVIEW

**Comment by 0xTim** [comment]
@ptoffy can you merge in the latest changes from `main` - that should trigger CI

**Comment by ptoffy** [comment]
@0xTim I think the tests are failing because the installed version of swift on CI is 5.2 and 5.4

**Comment by 0xTim** [comment]
@ptoffy yeah, now the CI is in your fork you should be able to change `swift:5.2-xenial` to `swift:5.5` etc

**Comment by ptoffy** [comment]
@0xTim have you had any time to review this?

File: Tests/LeafErrorMiddlewareTests/DefaultLeafErrorMiddlewareTests.swift [hunk 1]
```diff
@@ -143,57 +137,33 @@ class DefaultLeafErrorMiddlewareTests: XCTestCase {
...
-        XCTAssertEqual(viewRenderer.leafPath, "404")
-    }
-    
-    func testFutureNonAbort404IsCaughtCorrectly() throws {
-        let response = try app.getResponse(to: "/future404NoAbort")
-        XCTAssertEqual(response.status, .notFound)
-        XCTAssertEqual(viewRenderer.leafPath, "404")
-    }
-    
-    func testThatFuture403IsCaughtCorrectly() throws {
```
Comment (0xTim) [inline Tests/LeafErrorMiddlewareTests/DefaultLeafErrorMiddlewareTests.swift:165]: Could you add a test to ensure that 403s are caught? I'm sure they are but it was a previous issue so just want to make sure we don't add any regressions in the future!

File: Tests/LeafErrorMiddlewareTests/DefaultLeafErrorMiddlewareTests.swift [hunk 2]
```diff
@@ -1,118 +1,112 @@
...
-            router.get("unauthorized") { req -> EventLoopFuture<Response> in
+            router.get("unauthorized") { _ -> Response in
                 throw Abort(.unauthorized)
             }
-            
-            router.get("future404") { req -> EventLoopFuture<Response> in
-                return req.eventLoop.future(error: Abort(.notFound))
-            }
-            
-            router.get("future403") { req -> EventLoopFuture<Response> in
```
Comment (0xTim) [inline Tests/LeafErrorMiddlewareTests/DefaultLeafErrorMiddlewareTests.swift:53]: As below, could you add a 403 test here? Thanks!

File: Tests/LeafErrorMiddlewareTests/CustomGeneratorTests.swift [hunk 3]
```diff
@@ -139,36 +127,18 @@ class CustomGeneratorTests: XCTestCase {
...
 
-    func testFutureNonAbort404IsCaughtCorrectly() throws {
-        let response = try app.getResponse(to: "/future404NoAbort")
+    func testNonAbort404IsCaughtCorrectly() throws {
+        let response = try app.getResponse(to: "/404")
         XCTAssertEqual(response.status, .notFound)
         XCTAssertEqual(viewRenderer.leafPath, "404")
     }
 
-    func testThatFuture403IsCaughtCorrectly() throws {
```
Comment (0xTim) [inline Tests/LeafErrorMiddlewareTests/CustomGeneratorTests.swift:160]: I don't think we have a 403 test in this file, could you add one? Thanks!

**Review by 0xTim — CHANGES_REQUESTED** [review]
This is great thanks! There's just a couple of missing tests and we're good to go!

**Review by 0xTim — APPROVED** [review]
Thanks!

**Pull request merged**